### ISSUE TITLE
perf: train-every-N and a learner thread with a frozen policy

### ISSUE THREAD

**Issue by bmaltais** [bmaltais/slbot#11]
## Summary

Follow-up to #3. Overlapping `optimize_model()` with `env.step()` hides train cost **only while train ≤ env.step**. On CPU (and sometimes GPU) a 128-sample HybridDuelingDQN update is still slower than the worker step, so the overshoot delays the next action. Live snakes keep flying on the last command for that leftover time.

This issue covers the two remaining options from #3: train every N ticks, and a learner thread with a frozen inference policy.

## Why overlap is not enough

After #3 the tick is:

```
select_action → step_async → optimize_model() → step_wait → remember
```

Wall time is `max(env.step, train)`. If train is 150ms and env.step is 50ms, you still add 100ms of control latency.

## Approach A — train every N ticks

Nature DQN trained every 4 env steps (with action-repeat 4). We already have `frame_skip=3` and currently train **every** env step.

- Add `opt.train_every` (default 4, or 1 to preserve current learning)
- Only call `optimize_model()` when `batch_count % train_every == 0`
- Still overlap that call with `step_async` / `step_wait`

Cheap. Changes learning dynamics (fewer gradient steps per env step). Tune against a short run before making 4 the default.

## Approach B — learner thread + frozen policy

`select_action` and `optimize_model` share `policy_net`. A background train thread **with a lock around the net** still stalls inference for the whole backward.

Correct design:

1. **`inference_net`** — eval copy used only by `select_action` on the training thread. Never has gradients.
2. **`policy_net` / `target_net`** — trained on a learner thread.
3. Periodic weight publish: `inference_net.load_state_dict(policy_net.state_dict())` (or swap references) on a cadence that does not block the tick. Copying 160×160 HybridDuelingDQN is small compared to a backward, but do it between ticks or via a ready buffer.
4. If a train step is still running, **skip** the next one rather than queueing (drop-oldest). Agents never wait on AdamW.
5. CUDA: one device, no concurrent use of the same module from two threads. Learner owns `policy_net`; the tick only reads `inference_net`.

This is the complete “training cannot delay an action” fix.

## Suggested sequence

1. Land #3 overlap (if not already).
2. Add `train_every` behind config (A). Measure tick ms with N=1 vs N=4.
3. If ticks are still train-bound, implement B.

## Out of scope

- Do not cut `frame_skip` sleep
- Do not change Selenium/CDP in this issue (#4, #6)
